### PR TITLE
create Python __repr__ methods for QgsPoint and QgsPointXY

### PR DESCRIPTION
**Pull request by 3nids**
useful for quickly printing in the console or in magic *First Aid Kit* plugin

screen shot 2018-06-04 at 14 05 30

in QGIS 2.18 (will backport the Python code only)
image

### CODE REVIEW

**Comment by 3nids** [comment]
if everyone's happy with this, I will complete the other geometry classes

**Comment by Gustry** [comment]
From a python developer point of view, I'm of course happy. So so nice! Thanks

**Review by nyalldawson — COMMENTED** [review]
Looks brilliant! How about testing these too? Would be good to flag if it's ever broken by sipify changes.

**Comment by 3nids** [comment]
@nyalldawson any idea where you put these tests? in a dedicated Python test? or in a global for geometry? per class? (you probably know better than I do where to head...)

**Comment by nyalldawson** [comment]
Maybe put them in a dedicated test - that'll make it easy to extend as new ones are added

**Comment by nirvn** [comment]
@3nids , nice, looking especially good now that the console has a dedicated color for numbers in its scheme :)

**Comment by m-kuhn** [comment]
We've been waiting too long for this!

File: tests/src/python/test_python_repr.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+import sip
+
+start_app()
+
+
+class TestCoreAdditions(unittest.TestCase):
+
+    def TestQgsPointRepr(self):
+        p = QgsPoint(123.456, 987.654, 100)
+        print(p)
```
Comment (nyalldawson) [inline tests/src/python/test_python_repr.py]: Wouldn't it be better to actually check the result?

File: python/core/auto_additions/qgscircle.py [hunk 2]
```diff
@@ -0,0 +1,2 @@
+# The following has been generated automatically from src/core/geometry/qgscircle.h
+QgsCircle.__repr__ = lambda self: '<QgsPoint {}>'.format(self.toString())
```
Comment (nyalldawson) [inline python/core/auto_additions/qgscircle.py]: QgsCircle

**Comment by nyalldawson** [comment]
@3nids what's the target milestone here? hoping to have in for 3.2?

**Comment by 3nids** [comment]
moving them to SIP definition rather than Python injections, fixing the test and yep merge for 3.2

File: python/core/auto_generated/geometry/qgscircle.sip.in [hunk 3]
```diff
@@ -255,7 +255,7 @@ Returns true if the circle contains the ``point``.
 
     SIP_PYOBJECT __repr__();
 %MethodCode
-    QString str = QString( "<QgsCircle %1>" ).arg( sipCpp->toString() );
+    QString str = QString( "<QgsCircle: %1>" ).arg( sipCpp->toString() );
```
Comment (m-kuhn) [inline python/core/auto_generated/geometry/qgscircle.sip.in]: QStringLiteral

**Comment by nyalldawson** [comment]
Awesome.

**Pull request merged**